networks:
  reverse-proxy:
    external: true

services:
  portainer:
    image: portainer/portainer-ce:2.28.0-alpine@sha256:ff2b4387c1a8c89b92dc58bc50886365705e3bbb0d53b5b248cc3501b978db5a
    container_name: portainer
    restart: unless-stopped
    cpus: 0.1
    mem_limit: 128m
    # user: ${UID}:${GID}
    expose:
      - 9000
    networks:
      - reverse-proxy
    healthcheck:
      test: ["CMD", "wget", "--spider", "portainer:9000"]
      interval: 30s
      timeout: 10s
      retries: 3
      start_period: 10s
      start_interval: 5s
    volumes:
      - ~/home-server/portainer/data:/data:rw
      - /var/run/docker.sock:/var/run/docker.sock:ro
    environment:
      - TZ=${TZ}
    labels:
      - traefik.enable=true
      - traefik.http.routers.portainer.entrypoints=websecure
      - traefik.http.routers.portainer.rule=Host(`portainer.${DOMAIN_NAME}`)
      - traefik.http.routers.portainer.middlewares=chain-no-auth@file
      - traefik.http.services.portainer.loadbalancer.server.port=9000
